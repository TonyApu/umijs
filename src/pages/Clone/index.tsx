import { connect } from '@umijs/max';
import { Button, Form, Typography } from 'antd';
import { createStructuredSelector } from 'reselect';

import {
  fetchBreakfastAction,
  fetchCategoryAction,
  fetchDesertAction,
  fetchLunchAction,
  fetchReservationAction,
  fetchRestaurantAction,
  submitFormAction,
} from '../Admin/store/actions';
import {
  selectorBreakfast,
  selectorCategory,
  selectorDesert,
  selectorLunch,
  selectorRestaurant,
} from '../Admin/store/selectors';
import styles from './style.admin.scss';

const MockComponent: React.FC = (props: any) => {
  const { Title } = Typography;
  const [form] = Form.useForm();

  const onFinish = (values: any) => {
    props.submitForm(values);
  };

  return (
    <div className={styles.container}>
      <div className={styles.titleWrapper}>
        <Title level={2} className={styles.title}>
          Page admin
        </Title>
      </div>
      <Form form={form} layout="vertical" onFinish={onFinish}>
        <div className={styles.buttonWrapper}>
          <Button>Delete Screen (CTRL + R)</Button>
          <Button
            type="primary"
            className={styles.saveButton}
            htmlType="submit"
          >
            Save (CTRL + S)
          </Button>
        </div>
      </Form>
    </div>
  );
};

const mapStateToProps = createStructuredSelector({
  restaurant: selectorRestaurant,
  category: selectorCategory,
  breakfast: selectorBreakfast,
  lunch: selectorLunch,
  desert: selectorDesert,
});

export const mapDispatchToProps = (dispatch: any) => {
  return {
    fetchCategory: (id: string) => dispatch(fetchCategoryAction(id)),
    fetchRestaurant: () => dispatch(fetchRestaurantAction()),
    fetchBreakfast: (category: string) =>
      dispatch(fetchBreakfastAction(category)),
    fetchLunch: (category: string) => dispatch(fetchLunchAction(category)),
    fetchDesert: (category: string) => dispatch(fetchDesertAction(category)),
    fetchReservation: (id: string, resolve: any) =>
      dispatch(fetchReservationAction(id, resolve)),
    submitForm: (values: any) => dispatch(submitFormAction(values)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(MockComponent);
