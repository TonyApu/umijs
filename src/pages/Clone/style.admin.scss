$primary-color: #3486eb;

.container {
  .titleWrapper {
    min-height: 50px;
    .title {
      color: $primary-color;
    }
  }

  .buttonWrapper {
    display: flex;
    justify-content: flex-end;
    .saveButton {
      margin-left: 10px;
      background-color: $primary-color;
    }
  }

  .sectionTitle {
    background: $primary-color;
    padding: 5px;
    margin-top: 10px;
    h5 {
      margin-bottom: 0;
      color: white;
    }
  }

  .rowWrapper {
    margin-top: 10px;
    .datePicker {
      width: 100%;
    }
  }

  .rowLoading {
    margin: 20px 0;
    margin-bottom: 20px;
    text-align: center;
    border-radius: 4px;
  }
}
