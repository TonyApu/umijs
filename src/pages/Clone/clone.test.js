import * as React from "react";
import renderer from 'react-test-renderer';
import Clone, { mapDispatchToProps } from './index';

jest.mock('@umijs/max', () => {
  return {
    ...jest.requireActual('@umijs/max'),
    connect: () => (Component) => Component,
  };
});

// jest.mock('react', () => ({
//   ...jest.requireActual('react'),
//   useEffect: (effect) => effect(),
// }));

// jest.mock('ahooks', () => ({
//   ...jest.requireActual('ahooks'),
//   useUpdateEffect: (effect) => effect(),
// }));

jest.mock('../Admin/store/selectors', () => ({
  selectorRestaurant: jest.fn().mockReturnValue([
    {
      value: 'res1',
      label: 'Bar & Grill',
    },
    {
      value: 'res2',
      label: 'Gordan Ramsay Burger',
    },
  ]),
  selectorBreakfast: jest.fn().mockReturnValue([
    {
      value: 'egg',
      label: 'Egg',
    },
    {
      value: 'classic',
      label: 'Classic',
    },
  ]),
  selectorCategory: jest.fn().mockReturnValue(''),
  selectorLunch: jest.fn().mockReturnValue(''),
  selectorDesert: jest.fn().mockReturnValue(''),
  selectorReservation: jest.fn().mockReturnValue(''),
}));

jest.mock('antd', () => {
  const antd = jest.requireActual('antd');
  const Typography = ({ children }) => {
    return <typography>{children}</typography>;
  };
  Typography.Title = ({ children, ...otherProps }) => {
    return <title {...otherProps}>{children}</title>;
  };

  const Form = ({ children, ...otherProps  }) => {
    return <form {...otherProps}>{children}</form>;
  };

  // Form.Item = ({ children, ...otherProps }) => {
  //   return <item {...otherProps}>{children}</item>;
  // };

  return {
    ...antd,
    Typography,
    Form: {
      ...Form,
      useForm: () => [{
        setFieldValue: jest.fn(),
        submit: jest.fn(),
        resetFields: jest.fn()
      }],
    },
  };
});

describe('App component', () => {
  it('mock fetchRestaurant', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).fetchRestaurant();
    expect(dispatch.mock.calls[0][0]).toEqual({ type: 'menu/fetchRestaurant' });
  });

  it('mock fetchCategory', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).fetchCategory();
    expect(dispatch.mock.calls[0][0]).toEqual({ type: 'menu/fetchCategory' });
  });

  it('mock fetchBreakfast', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).fetchBreakfast();
    expect(dispatch.mock.calls[0][0]).toEqual({ type: 'menu/fetchBreakfast' });
  });

  it('mock fetchLunch', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).fetchLunch();
    expect(dispatch.mock.calls[0][0]).toEqual({ type: 'menu/fetchLunch' });
  });

  it('mock fetchDesert', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).fetchDesert();
    expect(dispatch.mock.calls[0][0]).toEqual({ type: 'menu/fetchDesert' });
  });

  it('mock fetchReservation', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).fetchReservation();
    expect(dispatch.mock.calls[0][0]).toEqual({
      type: 'menu/fetchReservation',
    });
  });

  it('mock submitForm', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).submitForm();
    expect(dispatch.mock.calls[0][0]).toEqual({ type: 'menu/submitForm' });
  });

  it('matches the snapshot', () => {
    const baseProps = {
      fetchRestaurant: jest.fn(),
      fetchCategory: jest.fn(),
      fetchBreakfast: jest.fn(),
      fetchLunch: jest.fn(),
      fetchDesert: jest.fn(),
      fetchReservation: jest.fn(),
    };
    const tree = renderer.create(<Clone {...baseProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
