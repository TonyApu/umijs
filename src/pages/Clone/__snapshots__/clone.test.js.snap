// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="titleWrapper"
  >
    <title
      className="title"
      level={2}
    >
      Page admin
    </title>
  </div>
  <div
    className="buttonWrapper"
  >
    <button
      className="ant-btn css-dev-only-do-not-override-1g853jt ant-btn-default"
      disabled={false}
      onClick={[Function]}
      style={{}}
      type="button"
    >
      <span>
        Delete Screen (CTRL + R)
      </span>
    </button>
    <button
      className="ant-btn css-dev-only-do-not-override-1g853jt ant-btn-primary saveButton"
      disabled={false}
      onClick={[Function]}
      style={{}}
      type="submit"
    >
      <span>
        Save (CTRL + S)
      </span>
    </button>
  </div>
</div>
`;
